src/pages/DatasetCatalog.js

```diff
@@ -301,6 +301,17 @@ export default function DatasetCatalog(props) {
     }
   };
 
+  const isEditDatasetEnabled = (dataset) => {
+    if (currentUser.isAdmin) {
+      return true;
+    }
+    // Chairpersons can only edit datasets they have direct access to via their DACs
+    if (currentUser.isChairPerson) {
+      return includes(dataset.dacId)(map('dacId')(filter({name: 'Chairperson'})(currentUser.roles)));
```

Comment:
`lodash/fp` has a `compose` function that can help make functions like this easier to read

https://hackernoon.com/function-composition-with-lodash-d30eb50153d1

So instead of nested arguments you would have something like...

```
return compose(
    includes(dataset.dacId),
    map('dacId')
    filter({name: 'Chairperson'})
)(currentUser.roles)
```
There's also a `flow` method that has the same end result, it just evaluates the functions differently...

```
return flow(
    filter({name: `Chairperson`}),
    map('dacId'),
    includes(dataset.dacId)
)(currentUser.roles)
```

Basically `flow` executes the innermost function first (listed as the first arg in flow) and applies the result to the outer function (subsequent function in the list)

I don't really have a preference as to what to use, so feel free to pick whichever if you so choose.

Reply:
I usually use `flow`, good call! 👍🏽